src/langchain_google_spanner/graph_store.py

```diff
@@ -0,0 +1,1176 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import datetime
+import re
+import string
+from abc import abstractmethod
+from typing import Any, Dict, List, Optional, Tuple, Type
+
+from google.cloud import spanner
+from google.cloud.spanner_v1 import param_types
+from langchain_community.graphs.graph_document import GraphDocument, Node, Relationship
+from langchain_community.graphs.graph_store import GraphStore
+from requests.structures import CaseInsensitiveDict
+
+
+def to_identifier(s: str) -> str:
```

Comment:
Can we move this method in `GraphDocumentUtility`?  as we are keeping `fixup_identifier` method there as well, both methods looks similar to me.

Reply:
Done!